frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx

```diff
@@ -441,9 +433,9 @@ export class UnconnectedDataSelector extends Component {
     }
   }
 
-  isSearchLoading = () => {
-    const { models, metrics, allLoading } = this.props;
-    return models == null || metrics == null || allLoading;
+  isLoading = () => {
+    const { models, metrics, loading } = this.props;
+    return models == null || metrics == null || loading;
```

Comment:
This is the fix: changing `allLoading` to `loading`.

It partially reverts this change: https://github.com/metabase/metabase/pull/50535/files#r1862678786

The old Entity Framework bug @ranquild was referring to has since been fixed (see [here](https://github.com/metabase/metabase/pull/51341/files#diff-186f83b598be6be0b89457e8b1f55b33d316e14d798b13f7094b0ede4413c843R290)).

`loading` works as expected now (`allLoading` still does not).

![image](https://github.com/user-attachments/assets/5291265a-9bc9-4478-9ad8-40c98588ec1b)

Reply:
This is incorrect since `loading` indicates the status of the last loader. Here we need at least 2, e.g. both models & metrics. You could potentially replace the second loader, pass `limit: 0`, and use `available_models` instead.

Reply:
25f61ef9cf2f015ad15470b91fffd16f7a7c7597

Reply:
@kamilmielnik you most likely don't need to add `rawData`, there should be `metadata` available

Reply:
db6919326922278103b5dc9e4070e337de2bea55